### PR TITLE
Navbar: Toggle dropdown menus of navbar with Enter,Space and close with Escape

### PR DESCRIPTION
**Pull request by ElishaSamPeterPrabhu**
## Description

- The addMenu is set via slot method and is part of the modus-dropdown. So the keydown features are added from modus-dropdown to navbar-button-lust to the modus-navbar
- Space and enter toggle the menu and escape hides it

References #
#1628 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by cjwinsor — CHANGES_REQUESTED** [review]

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx [hunk 1]
```diff
@@ -329,6 +328,21 @@ export class ModusNavbar {
...
+  buttonMenuKeydownHandler(event: KeyboardEvent, button: ModusNavbarButton) {
+    if (event.code == 'Enter' || event.code == 'Space') {
+      event.preventDefault();
+      this.buttonClick.emit(button.id);
+      if (this.openButtonMenuId == button.id) {
+        this.hideMenus();
+      } else {
+        this.hideMenus();
+        this.openButtonMenuId = button.id;
+      }
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-navbar/modus-navbar.tsx]: Refactored

File: stencil-workspace/src/components/modus-navbar/modus-navbar.tsx [hunk 2]
```diff
@@ -329,6 +328,21 @@ export class ModusNavbar {
       this.openButtonMenuId = button.id;
     }
   }
+  buttonMenuKeydownHandler(event: KeyboardEvent, button: ModusNavbarButton) {
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-navbar/modus-navbar.tsx]: Renamed

File: stencil-workspace/src/components/modus-dropdown/modus-dropdown.tsx [hunk 3]
```diff
@@ -86,6 +100,22 @@ export class ModusDropdown {
...
+        this.visible = false;
+      }
+
+      if (!this.visible) {
+        this.dropdownClose.emit();
+      } else {
+        this.dropdownToggleClicked = true;
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (ElishaSamPeterPrabhu) [reply stencil-workspace/src/components/modus-dropdown/modus-dropdown.tsx:119]: There is room for refactoring with the `hideDropdown()` but `showDropdown()` has different cases with the current logic, hence refactored with `hideDropdown()`.

**Comment by cjwinsor** [comment]
@ElishaSamPeterPrabhu Can you add some tests around this functionality, to ensure the key handling is as we expect?

**Comment by ElishaSamPeterPrabhu** [comment]
> @ElishaSamPeterPrabhu Can you add some tests around this functionality, to ensure the key handling is as we expect?

Added test cases for the added keyboard events

**Comment by cjwinsor** [comment]
@ElishaSamPeterPrabhu Can you see if you can find a solution for closing the menus when you tab out of them.  Currently all the keyboard interactions you added are working, but if i open the menu with the keyboard, then tab through it and onto the next, i can open that one and the previous is not closed.

**Comment by ElishaSamPeterPrabhu** [comment]
> @ElishaSamPeterPrabhu Can you see if you can find a solution for closing the menus when you tab out of them. Currently all the keyboard interactions you added are working, but if i open the menu with the keyboard, then tab through it and onto the next, i can open that one and the previous is not closed.
> 
> image

Fixed

**Review by cjwinsor — APPROVED** [review]

**Pull request merged**